### PR TITLE
Merge GEOS-gtFV3 interface into develop

### PR DESCRIPTION
**Pull request by pchakraborty**
Build and run gtFV3 on demand

1. Code to build the GEOS-gtFV3 interface is activated by the cmake option `BUILD_GEOS_GTFV3_INTERFACE`
2. `BUILD_GEOS_GTFV3_INTERFACE` also sets the macro `RUN_GTFV3`, used in `FV_StateMod.F90`
3. All of the interface related code resides in the directory `geos-gtfv3`

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,83 @@
 esma_set_this ()
 
+option(BUILD_GEOS_GTFV3_INTERFACE "Build GEOS-gtFV3 interface" OFF)
```
Comment (tclune) [inline CMakeLists.txt:3]: @pchakraborty Can most of this new chunk be moved lower in the CMakefile.    I'm concerned that the typical user comes here just to add/delete a source file and now has to scan down.   If at least those sources could be moved up.

**Comment by mathomp4** [comment]
Also pinging @aoloso as I know he is looking at `FV_StateMod` at the moment. Just to keep him aware.

File: CMakeLists.txt (hunk 1 above)
Comment (mathomp4) [reply CMakeLists.txt:3]: Yeah. Might be best if the main `srcs` list was still at the top. That's probably what most people will be looking for in here...

Comment (mathomp4) [reply CMakeLists.txt:3]: Looks good to me. I'll let @tclune weigh in. :)

**Review by tclune — APPROVED** [review]

**Review by mathomp4 — APPROVED** [review]
I tested this and it seems to be zero-diff when not building with GTFV3 (which I don't know how to do)!

**Pull request merged**